### PR TITLE
Implements HTTPS support for hieraserver

### PR DESCRIPTION
**Pull request by flashvoid**
Hello

I suggest enabling hieraserver to run in HTTPS configuration since it is likely to be serving secrets.

### CODE REVIEW

File: hieraserver/rest/rest.go [hunk 1]
```diff
@@ -70,7 +81,25 @@ func startServer(cmd *cobra.Command, _ []string) {
...
-		err := http.ListenAndServe(addr+":"+strconv.Itoa(port), router)
+
+		server := &http.Server{
+			Addr:    addr + ":" + strconv.Itoa(port),
+			Handler: router,
+		}
+
+		var err error
+		var tlsConfig *tls.Config
+		tlsConfig, err = makeTLSconfig()
```
Comment (thallgren) [inline hieraserver/rest/rest.go]: Please skip preceding `var` declarations and use `:=`.

File: hieraserver/rest/rest.go [hunk 2]
```diff
@@ -122,3 +151,47 @@ func CreateRouter(ctx px.Context) http.Handler {
...
+func loadCertPool(pemFile string) (*x509.CertPool, error) {
+	data, err := ioutil.ReadFile(pemFile)
+	if err != nil {
+		return nil, err
+	}
+
+	certPool := x509.NewCertPool()
+	ok := certPool.AppendCertsFromPEM(data)
+	if !ok {
+		return nil, fmt.Errorf("Failed to load certificate %s", pemFile)
```
Comment (thallgren) [inline hieraserver/rest/rest.go]: Please use `%q` to quote the path as it's not uncommon that it contains spaces.

File: hieraserver/rest/rest.go (hunk 1 above)
Comment (flashvoid) [reply hieraserver/rest/rest.go]: done

File: hieraserver/rest/rest.go (hunk 2 above)
Comment (flashvoid) [reply hieraserver/rest/rest.go]: done

**Pull request merged**